fakestorage/object.go

```diff
@@ -323,38 +325,63 @@ type ListOptions struct {
 	StartOffset              string
 	EndOffset                string
 	IncludeTrailingDelimiter bool
+
+	PageToken string
+	MaxKeys   *int
```

Comment:
Do you think you could send the pagination changes separately? Also, ideally we'd introduce them without breaking the signature of `ListObjects` and `ListObjectsWithOptions` (which maybe means we need a third method, which is definitely not great). We can make that method private though and have `ListObjects` and `ListObjectsWithOptions` always return all objects.